packages/docs/src/components/DocSiteNav.tsx

```diff
@@ -1,9 +1,87 @@
 import React from 'react';
+import { VerticalNav } from '@cmsgov/design-system';
+import { VerticalNavItemProps } from '@cmsgov/design-system/src/components/VerticalNav/VerticalNavItem';
+import { useStaticQuery, graphql } from 'gatsby';
 
+interface NavItem {
+  id: string;
+  name: string;
+  relativeDirectory: string;
+  relativePath: string;
+}
+
+/**
+ * DocSiteNav
+ * queries all MDX files and generates the proper format to create a vertical nav
+ * @returns {React Element}
+ * @todo figure out which item is currently selected and mark & expand appropriately
+ * @todo Determine URL string for each item once content is integrated
+ */
 const DocSiteNav = () => {
+  const data = useStaticQuery(graphql`
+    query SiteNavQuery {
+      allFile(
+        filter: { absolutePath: {}, extension: { eq: "mdx" } }
+        sort: { fields: [relativeDirectory, name] }
+      ) {
+        nodes {
+          id
+          name
+          relativeDirectory
+          relativePath
+        }
```

Comment:
I was playing around with the GraphQL playground (didn't try Sarah's `allFile` -> `allMdx` idea though) and saw that there was a way to group the results. It wouldn't take away the need to do some kind of transformation to get valid `VerticalNavItemProps`, but it could possibly reduce the aggregating code:
```graphql
query SiteNavQuery {
  allFile(sort: {fields: [relativeDirectory, name]}) {
    group(field: relativeDirectory) {
      fieldValue
      edges {
        node {
          id
          name
          relativeDirectory
          relativePath
        }
      }
    }
  }
}
```

In this case, the result will have a `fieldValue` of `""` for the `intro` file where there is no relative directory, and all others will have the directory name inside `fieldValue`.